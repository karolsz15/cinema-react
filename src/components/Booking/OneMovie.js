import React from 'react';
import PropTypes from 'prop-types';

const OneMovie = ({ poster, title, summary, clicked1, clicked2, hour1, hour2 }) => {
  return (
    <>
      <div className="booking_object">
        <img src={poster} alt="..." style={{ margin: '.5em' }} />
        <div className="media-body">
          <h5 style={{ margin: '.5em' }}>{title}</h5>
          {summary}
          <div>
            <button style={{ margin: '.5em' }} onClick={clicked1} className="btn btn-secondary">
              {hour1}
            </button>
            <button style={{ margin: '.5em' }} onClick={clicked2} className="btn btn-secondary">
              {hour2}
            </button>
          </div>
        </div>
      </div>
      <hr />
    </>
  );
};

OneMovie.propTypes = {
  poster: PropTypes.string,
  title: PropTypes.string,
  summary: PropTypes.string,
  clicked1: PropTypes.func,
  clicked2: PropTypes.func,
  hour1: PropTypes.string,
  hour2: PropTypes.string,
};

export default OneMovie;
