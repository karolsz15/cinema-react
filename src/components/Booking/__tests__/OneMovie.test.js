import React from 'react';
import { shallow } from 'enzyme';
import OneMovie from '../OneMovie';

const wrapper = shallow(<OneMovie />);

describe('OneMovie component', () => {
  it('should render the OneMovie correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
