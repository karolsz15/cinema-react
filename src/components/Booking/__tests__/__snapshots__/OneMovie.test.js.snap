// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneMovie component should render the OneMovie correctly 1`] = `
<Fragment>
  <div
    className="booking_object"
  >
    <img
      alt="..."
      style={
        Object {
          "margin": ".5em",
        }
      }
    />
    <div
      className="media-body"
    >
      <h5
        style={
          Object {
            "margin": ".5em",
          }
        }
      />
      <div>
        <button
          className="btn btn-secondary"
          style={
            Object {
              "margin": ".5em",
            }
          }
        />
        <button
          className="btn btn-secondary"
          style={
            Object {
              "margin": ".5em",
            }
          }
        />
      </div>
    </div>
  </div>
  <hr />
</Fragment>
`;
